obp-api/src/main/scala/code/api/v4_0_0/OBPAPI4_0_0.scala

```diff
@@ -36,403 +36,44 @@ import code.api.v2_1_0.APIMethods210
-    Implementations4_0_0.createTransactionRequest :: 
-    Implementations4_0_0.answerTransactionRequestChallenge ::
-    Implementations4_0_0.getDynamicEntities ::
-    Implementations4_0_0.createDynamicEntity ::
-    Implementations4_0_0.updateDynamicEntity ::
-    Implementations4_0_0.deleteDynamicEntity ::
-    Implementations4_0_0.genericEndpoint ::
-    Implementations4_0_0.resetPasswordUrl ::
-    Implementations4_0_0.root ::
-    Implementations4_0_0.getEntitlements ::
-    Nil
+  // Possible Endpoints from 4.0.0, exclude one endpoint use - method,exclude multiple endpoints use -- method,
+  // e.g getEndpoints(Implementations4_0_0) -- List(Implementations4_0_0.genericEndpoint, Implementations4_0_0.root)
+  val endpointsOf4_0_0 = getEndpoints(Implementations4_0_0) - Implementations4_0_0.genericEndpoint
   
   def allResourceDocs = MockerConnector.doc ++
                         Implementations4_0_0.resourceDocs ++
-                        Implementations3_1_0.resourceDocs ++
-                        Implementations3_0_0.resourceDocs ++
-                        ImplementationsCustom3_0_0.resourceDocs ++
-                        Implementations2_2_0.resourceDocs ++
-                        Implementations2_1_0.resourceDocs ++
-                        Implementations2_0_0.resourceDocs ++
-                        Implementations1_4_0.resourceDocs ++
-                        Implementations1_3_0.resourceDocs ++
-                        Implementations1_2_1.resourceDocs
+                        OBPAPI3_1_0.allResourceDocs
+
 
   def findResourceDoc(pf: OBPEndpoint): Option[ResourceDoc] = {
     allResourceDocs.find(_.partialFunction==pf)
   }
 
   // Filter the possible endpoints by the disabled / enabled Props settings and add them together
   val routes : List[OBPEndpoint] =
-    Implementations4_0_0.genericEndpoint ::
       Implementations4_0_0.root :: // For now we make this mandatory
-  getAllowedEndpoints(endpointsOf1_2_1, Implementations1_2_1.resourceDocs) :::
-  getAllowedEndpoints(endpointsOf1_3_0, Implementations1_3_0.resourceDocs) :::
-  getAllowedEndpoints(endpointsOf1_4_0, Implementations1_4_0.resourceDocs) :::
-  getAllowedEndpoints(endpointsOf2_0_0, Implementations2_0_0.resourceDocs) :::
-  getAllowedEndpoints(endpointsOf2_1_0, Implementations2_1_0.resourceDocs) :::
-  getAllowedEndpoints(endpointsOf2_2_0, Implementations2_2_0.resourceDocs) :::
-  getAllowedEndpoints(endpointsOf3_0_0, Implementations3_0_0.resourceDocs) :::
-  getAllowedEndpoints(endpointsOfCustom3_0_0, ImplementationsCustom3_0_0.resourceDocs) :::
-  getAllowedEndpoints(endpointsOf3_1_0, Implementations3_1_0.resourceDocs) :::
-  getAllowedEndpoints(endpointsOf4_0_0, Implementations4_0_0.resourceDocs)
+      getAllowedEndpoints(endpointsOf4_0_0, Implementations4_0_0.resourceDocs) ++
+      OBPAPI3_1_0.routes // ***** here the previous version routes added at last, so there no need exclude any previous endpoints, because they will be omit.
```

Comment:
the previous routes add here, no need exclude anyone, because version 4.0.0 endpoints lay before this line, so they have high priority, So we don't need to exclude endpoints as before.